mlflow/models/evaluation/validation.py

```diff
@@ -68,16 +75,22 @@ def __init__(
                 raise MetricThresholdClassException(
                     "`min_relative_change` parameter must be between 0 and 1."
                 )
-        if higher_is_better is None:
-            raise MetricThresholdClassException("`higher_is_better` parameter must be defined.")
-        if not isinstance(higher_is_better, bool):
-            raise MetricThresholdClassException("`higher_is_better` parameter must be a boolean.")
+        if higher_is_better is None and greater_is_better is None:
+            raise MetricThresholdClassException("`greater_is_better` parameter must be defined.")
+        if higher_is_better is not None and greater_is_better is not None:
+            raise MetricThresholdClassException(
+                "`higher_is_better` parameter must be None when `greater_is_better` is defined."
+            )
```

Comment:
Can we extend this to check if there's a mismatch in value? I.e one is False and one is True?. If there's a mismatch, we should throw.